Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM ubuntu:xenial as base
+
+# Install system requirements
+RUN apt update && \
+    # Global requirements
+    DEBIAN_FRONTEND=noninteractive apt install -y language-pack-en git build-essential software-properties-common curl git-core libxml2-dev libxslt1-dev libmysqlclient-dev libxmlsec1-dev libfreetype6-dev libssl-dev swig gcc g++ \
+    # openedx requirements
+    gettext gfortran graphviz libgraphviz-dev libffi-dev libfreetype6-dev libgeos-dev libjpeg8-dev liblapack-dev libpng-dev libsqlite3-dev libxml2-dev libxmlsec1-dev libxslt1-dev ntp pkg-config python3.5 python3-pip python3-dev \
+    -qy && rm -rf /var/lib/apt/lists/*
+
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+RUN ln -s /usr/bin/pip3 /usr/bin/pip
+RUN ln -s /usr/bin/python3 /usr/bin/python
+
+WORKDIR /edx/app/edx-platform/edx-platform
+
+COPY . /edx/app/edx-platform/edx-platform
+
+ENV PATH /edx/app/edx-platform/nodeenv/bin:${PATH}
+ENV PATH ./node_modules/.bin:${PATH}
+ENV CONFIG_ROOT /edx/etc/
+ENV PATH /edx/app/edx-platform/edx-platform/bin:${PATH}
+ENV SETTINGS production
+
+RUN pip install setuptools==39.0.1 pip==9.0.3
+RUN pip install -r requirements/edx/base.txt
+RUN pip install newrelic
```

Comment:
Shouldn't it be optional via an `EXTRA_DEPENDENCIES` build `ARG`?

Reply:
I would like to avoid build args if possible just to keep the interface as clean and simple as possible.

Or atleast I would like 'docker build .' to produce the artifact without additional build args. The way i've done that with the other apps is to make a separate build stage that included newrelic and base it off the first stage to preserve the 'docker build .' interface.

Yours is more flexible perhaps, maybe there is a best of both worlds here? I haven't played with build args a ton truth be told

Reply:
I agree that it shouldn't be included in the main Dockerfile target either way.

Reply:
I think this is good advice, I had been doing them as additional build stages but you are right, this is probably cleaner.  I will run this by the team and see what they think, as I would prefer you guys to not get NR when you docker build .

Reply:
Thanks for the feedback